### PR TITLE
303 reorder authors in metadata

### PR DESCRIPTION
**Pull request by ilya-zhuravel**
## Why

Drag and drop author cards to rearrange order in metadata is a design requirement

## What

Metadata editor is a scrollable container, and so to support autoscroll I used dom-autoscroller, which works well with dragula drag&drop library, so the SortableContainer implentation had to be refactored.

### CODE REVIEW

File: src/article/components/SortableContainerComponent.js [hunk 1]
```diff
@@ -17,91 +18,74 @@ class SortableComponent extends Component {
...
-      .on('dragstart', this._dragStart, this)
-      .on('dragover', this._dragOver, this)
-      .on('dragend', this._dragEnd, this);
-
-    const draggableChildren = this.props.children.map((child, index) => {
-      return $$(SortableComponent, {sortIndex: index}).append(child);
-    });
-
-    return sortableContainer.append(draggableChildren);
+  get direction () {
```
Comment (NuclearRedeye) [inline src/article/components/SortableContainerComponent.js]: Function could be named better, something like 'isHorizontal' or similar.

File: src/article/metadata/CardComponent.js [hunk 2]
```diff
@@ -26,8 +26,8 @@ export default class CardComponent extends Component {
           .append(label),
       );
     el.append(children);
-    el.on('mousedown', this._onMousedown);
-    el.on('click', this._onClick);
+    // el.on('mousedown', this._onMousedown);
```
Comment (NuclearRedeye) [inline src/article/metadata/CardComponent.js]: As this is on the CardComponent class, how does this impact other card instances?

File: src/article/metadata/CardComponent.js [hunk 3]
```diff
@@ -63,7 +63,7 @@ export default class CardComponent extends Component {
   }
 
   _onClick(e) {
-    domHelpers.stopAndPrevent(e);
+    // domHelpers.stopAndPrevent(e);
```
Comment (NuclearRedeye) [inline src/article/metadata/CardComponent.js]: Same here? Doe this cause any other issues for other instances of the CardComponent? If so, perhaps an idea might be to add a 'bubbles' prop, and wrap this up in an 'if (this.props.bubbles) { preventDefault() };'?

File: src/article/metadata/CardComponent.js (hunk 2 above)
Comment (NuclearRedeye) [reply src/article/metadata/CardComponent.js]: So I just tested this and now I can no longer select a card and right click to get the context menu, for example to delete a reference or keyword. So the behaviour has definitely changed. As such I think the behaviour here needs to switch depending on if it is a ScrollableContainer or not just to preserve the original behaviour.
We can review this again when more affordances are added to the UI to replace the missing functionality.

**Review by NuclearRedeye — CHANGES_REQUESTED** [review]
Tested fine, now that the build fix is in place. That's also made Travis happy. But there are some other tweaks that I think need to be made

File: src/article/metadata/CardComponent.js (hunk 3 above)
Comment (ilya-zhuravel) [reply src/article/metadata/CardComponent.js]: I can't find any issues with not stopping this event from bubbling up

**Review by NuclearRedeye — APPROVED** [review]

**Pull request merged**